File: app/src/main/java/com/ucb/eldroid/ecoconnect/MainActivity.java
package com.ucb.eldroid.ecoconnect;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.ucb.eldroid.ecoconnect.data.ApiService;
import com.ucb.eldroid.ecoconnect.data.models.LoginRequest;
import com.ucb.eldroid.ecoconnect.data.models.User;
import com.ucb.eldroid.ecoconnect.utils.RetrofitClient;

import java.util.HashMap;
import java.util.Map;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    private ApiService apiService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        apiService = RetrofitClient.getInstance().create(ApiService.class);

        // Example: Register User
        Map<String, String> registerData = new HashMap<>();
        registerData.put("name", "John Doe");
        registerData.put("email", "johndoe@example.com");
        registerData.put("password", "password123");
        registerData.put("password_confirmation", "password123");

        apiService.registerUser(registerData).enqueue(new Callback<Void>() {
            @Override
            public void onResponse(Call<Void> call, Response<Void> response) {
                if (response.isSuccessful()) {
                    Toast.makeText(MainActivity.this, "User registered successfully!", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, "Registration failed.", Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<Void> call, Throwable t) {
                Toast.makeText(MainActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });

        // Example: Login User
        Map<String, String> loginData = new HashMap<>();
        loginData.put("email", "johndoe@example.com");
        loginData.put("password", "password123");

        apiService.loginUser(loginData).enqueue(new Callback<Map<String, String>>() {
            @Override
            public void onResponse(Call<Map<String, String>> call, Response<Map<String, String>> response) {
                if (response.isSuccessful()) {
                    String token = response.body().get("token");
                    Toast.makeText(MainActivity.this, "Login successful! Token: " + token, Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, "Login failed.", Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<Map<String, String>> call, Throwable t) {
                Toast.makeText(MainActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/ucb/eldroid/ecoconnect/MainActivity.kt
package com.ucb.eldroid.ecoconnect

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.ucb.eldroid.ecoconnect.data.ApiService
import com.ucb.eldroid.ecoconnect.utils.RetrofitClient
import kotlinx.coroutines.launch
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    private lateinit var apiService: ApiService

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val apiService = RetrofitClient.instance.create(ApiService::class.java)

        lifecycleScope.launch {
            // Example: Register User
            val registerData = mapOf(
                "name" to "John Doe",
                "email" to "johndoe@example.com",
                "password" to "password123",
                "password_confirmation" to "password123"
            )

            try {
                val registerResponse = apiService.registerUser(registerData)
                if (registerResponse.isSuccessful) {
                    Toast.makeText(this@MainActivity, "User registered successfully!", Toast.LENGTH_SHORT).show()
                } else {
                    Toast.makeText(this@MainActivity, "Registration failed.", Toast.LENGTH_SHORT).show()
                }
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Error: ${e.message}", Toast.LENGTH_SHORT).show()
            }

            // Example: Login User
            val loginData = mapOf(
                "email" to "johndoe@example.com",
                "password" to "password123"
            )

            try {
                val loginResponse = apiService.loginUser(loginData)
                if (loginResponse.isSuccessful) {
                    val token = loginResponse.body()?.get("token")
                    Toast.makeText(this@MainActivity, "Login successful! Token: $token", Toast.LENGTH_SHORT).show()
                } else {
                    Toast.makeText(this@MainActivity, "Login failed.", Toast.LENGTH_SHORT).show()
                }
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Error: ${e.message}", Toast.LENGTH_SHORT).show()
            }
        }
    }
}